Forms/HTMLEditor/FileShortcodeProvider.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace SilverStripe\Forms\HtmlEditor;
```

Comment:
I don't think this is the right namespace – this shortcode handler should be part of the `SilverStripe\Assets` subsystem. If you didn't have `SilverStripe\Assets` you wouldn't have this shortcode provider available. This would be like putting the models of every SilverStripe module into the `SilverStripe\ORM` namespace.

Perhaps this can go into the `SilverStripe\Assets\ViewSupport` namespace, since it's a support class to assist the view?

Obviously you could argue: "if you want to use silverstripe\assets without the silverstripe WYSIWYG editor, then you've got the same problem in reverse". However, I think that we're more likely to require the WYSIWYG editor without assets than the reverse.